grpc-spring-boot-starter/src/main/java/org/lognet/springboot/grpc/autoconfigure/GRpcServerProperties.java

```diff
@@ -50,7 +50,7 @@
 
     /**
      * Number of seconds to wait for preexisting calls to finish before shutting down.
-     * A negative value is equivalent to an infinite grace period
+     * Zero or negative value is equivalent to an infinite grace period
```

Comment:
Zero means "no wait" and not infinite

Reply:
I was wrong that current behavior is no wait. The wait was infinite in awaiter thread.

Reply:
I think you were correct originally. The wait is infinite in the awaiter thread, but that is a way to keep the server running during normal operations, and has nothing to do with the JVM shutdown process. During JVM process, we were not calling `awaitTermination` and not waiting for any preexisting calls to finish before allowing the JVM to shut down. The value here sets how long to wait during shutdown, not during normal operations.